src/adv.py

```diff
@@ -33,19 +42,79 @@
 room['narrow'].n_to = room['treasure']
 room['treasure'].s_to = room['narrow']
 
+# Link items to rooms
+room['outside'].items = [items['sword']]
+room['foyer'].items = [items['helmet']]
+room['narrow'].items = [items['armour']]
+room['overlook'].items = [items['gem']]
+room['treasure'].items = [items['scroll'], items['treasure']]
+
 #
 # Main
 #
 
 # Make a new player object that is currently in the 'outside' room.
+player = Player("Player1", room['outside'])
 
-# Write a loop that:
-#
-# * Prints the current room name
-# * Prints the current description (the textwrap module might be useful here).
-# * Waits for user input and decides what to do.
-#
-# If the user enters a cardinal direction, attempt to move to the room there.
-# Print an error message if the movement isn't allowed.
-#
-# If the user enters "q", quit the game.
+while True: 
+    
+    player.details()
+    player.check_items()
+
+    try:
+        # Gets user input, normalizes it, creates list so we can parse 
+        cmd = input("\nEnter a command: ").lower().split()
+
+        # CASE: user entered 1 word (direction)
+        if len(cmd) == 1:
+
+            # starts with q, quit game (break out of REPL)
+            if cmd[0][0] == "q":
+                print("Game ended")
+                break
+
+            # starts with i, show inventory
+            if cmd[0][0] == "i":
+                player.show_inventory()
+                
+            # checks if the direction is a possible "move". Delegates correct move
+            attribute = f'{cmd[0][0]}_to'
+            if hasattr(player.current_room, attribute):
+                # set the new room
+                player.current_room =  getattr(player.current_room, attribute)
+            else:
+                print("Wrong way! Try again, hint: enter n, s, e, w")
+                continue
+
+        # CASE: user entered 2 words (some action followed by an item)
+        elif len(cmd) == 2:
```

Comment:
This is brilliant! I would love to see how you thought through your code here, might hit you up with a zoom!